### PR TITLE
Trim new line on API Key

### PR DESCRIPTION
**Pull request by achntrl**
### What does this PR do?

Trim new line on API key

### Motivation

Sometime, we receive API key with trailing new line and it breaks silently break the log ingestion.

### CODE REVIEW

**Review by tmichelet — DISMISSED** [review]
Thanks for fixing this. LGTM but let's ask someone from agent-core to also review

File: pkg/config/config.go [hunk 1]
```diff
@@ -434,9 +434,17 @@ func Load() error {
 	}
 
 	loadProxyFromEnv()
+	trimAPIKeyTrailingNewline()
 	return nil
 }
 
+// Avoid log ingestion breaking because of a newline in the API key
+func trimAPIKeyTrailingNewline() {
```
Comment (ajacquemot) [inline pkg/config/config.go]: nits: WDYT about `sanitizeAPIiKey` instead of `trimAPIKeyTrailingNewline`

File: pkg/config/config.go [hunk 2]
```diff
@@ -434,9 +434,17 @@ func Load() error {
...
+	if Datadog.IsSet("api_key") {
```
Comment (ajacquemot) [inline pkg/config/config.go]: I am not sur it's worth checking the `api_key`, you could do something like:
```
api_key := Datadog.GetString("api_key")
Datadog.Set("api_key", strings.TrimSpace(api_key) )
```

File: pkg/config/config_test.go [hunk 3]
```diff
@@ -344,3 +344,13 @@ func TestLoadProxyEmptyValuePrecedence(t *testing.T) {
...
 	Datadog.Set("proxy", nil)
 }
+
+func TestTrimAPIKeyTrailingNewline(t *testing.T) {
+	Datadog.Set("api_key", "foo")
+	trimAPIKeyTrailingNewline()
+	assert.Equal(t, "foo", Datadog.GetString("api_key"))
+
+	Datadog.Set("api_key", "foo\n")
+	trimAPIKeyTrailingNewline()
```
Comment (ajacquemot) [inline pkg/config/config_test.go]: Can you add more tests please, with leading/trailing white space, EOL, etc

File: .gitignore [hunk 4]
```diff
@@ -67,3 +67,6 @@ cmd/agent/android/app/build
 
 # local copy of device signing cert
 platform.pk8
+
+# virtualenv
+venv/
```
Comment (ajacquemot) [inline .gitignore]: I think you'd better open a new PR for this

**Review by ajacquemot — APPROVED** [review]

**Review by arbll — APPROVED** [review]
:shipit:

**Pull request merged**